openwisp_radius/api/permissions.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from django.utils.translation import gettext_lazy as _
+from rest_framework.exceptions import APIException, PermissionDenied
+from rest_framework.permissions import BasePermission
+
+logger = logging.getLogger(__name__)
+
+
+class HasSmsVerification(BasePermission):
+    def has_permission(self, request, view):
+        """
+        raises PermissionDenied if organization has disabled sms_verification
+        """
+        organization = getattr(view, 'organization')
+        verification = self.has_sms_verification(view, organization)
+        if not verification:
+            message = _(
+                f'View {view.__class__.__name__} is being accessed for organization '
+                f'{organization.name} but SMS verification is disabled for '
+                'this organization.'
+            )
+            logger.warning(message)
+            raise PermissionDenied(message)
+        else:
+            return True
+
+    def has_sms_verification(self, view, org):
+        try:
+            verification = org.radius_settings.sms_verification
+        except Exception as e:
+            # Raise exception in case an attribute is missing from org
+            logger.error(e)
+            raise APIException(e)
```

Comment:
Never leak internal system information to users.

Use a generic error message instead.

In this case, there's already another part of the system which does a similar check, we could reuse the same string: https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/api/serializers.py#L335-L340
You could make the string a variable and use it in both places, so we avoid repetition.

Reply:
I just reused the same string. Is it verbose this way, should I use it as a variable instead? If yes, which file would be suitable for storing it in that case?